lib/util/cache.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * @see https://www.rfc-editor.org/rfc/rfc9111.html#name-cache-control
+ * @see https://www.iana.org/assignments/http-cache-directives/http-cache-directives.xhtml
+ *
+ * @typedef {{
+ *   'max-stale'?: number;
+ *   'min-fresh'?: number;
+ *   'max-age'?: number;
+ *   's-maxage'?: number;
+ *   'stale-while-revalidate'?: number;
+ *   'stale-if-error'?: number;
+ *   public?: true;
+ *   private?: true;
+ *   'no-store'?: true;
+ *   'no-cache'?: true | string[];
+ *   'must-revalidate'?: true;
+ *   'proxy-revalidate'?: true;
+ *   immutable?: true;
+ *   'no-transform'?: true;
+ *   'must-understand'?: true;
+ *   'only-if-cached'?: true;
+ * }} CacheControlDirectives
+ *
+ * @param {string} header
+ * @returns {CacheControlDirectives}
+ */
+function parseCacheControlHeader (header) {
+  /**
+   * @type {import('../util/cache.js').CacheControlDirectives}
+   */
+  const output = {}
+
+  const directives = header.toLowerCase().split(',')
+  for (let i = 0; i < directives.length; i++) {
+    const directive = directives[i]
+    const keyValueDelimiter = directive.indexOf('=')
+
+    let key
+    let value
+    if (keyValueDelimiter !== -1) {
+      key = directive.substring(0, keyValueDelimiter).trim()
+      value = directive
+        .substring(keyValueDelimiter + 1, directive.length)
+        .trim()
+        .toLowerCase()
+    } else {
+      key = directive.trim()
+    }
+
+    switch (key) {
+      case 'min-fresh':
+      case 'max-stale':
+      case 'max-age':
+      case 's-maxage':
+      case 'stale-while-revalidate':
+      case 'stale-if-error': {
+        const parsedValue = parseInt(value, 10)
+        if (isNaN(parsedValue)) {
```

Comment:
Maybe better use Number.isNaN, as we should already have a number at this point.
Btw. you could also do for an NaN check

```suggestion
        if (parsedValue !== parsedValue) {
```